frappe/core/doctype/communication/communication.py

```diff
@@ -161,8 +161,22 @@ def set_sender_full_name(self):
 				sender_name, sender_email = parse_addr(self.sender)
 				if sender_name == sender_email:
 					sender_name = None
+
 				self.sender = sender_email
-				self.sender_full_name = sender_name or frappe.db.exists("Contact", {"email_id": sender_email}) or sender_email
+				self.sender_full_name = sender_name
```

Comment:
Use can use this function [get_contact_name](https://github.com/frappe/frappe/blob/639da6ec549a60b96ff80fb996dbe8a36c5486fb/frappe/contacts/doctype/contact/contact.py#L208) to get the Contact Name from child tables containing all the emails.

Reply:
@hrwX get_contact_name returns the `name` of the Contact and not the actual `fullname` which is built from `first_name` and `last_name`